nntrainer/tensor/matrix_transpose_neon/matrix_transpose_kernels_neon.h

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Sungsik Kong <ss.kong@samsung.com>
+ *
+ * @file   matrix_transpose_kernels_neon.h
+ * @date   09 May 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Sungsik Kong <ss.kong@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  These are internal util functions for transposing matrix with NEON
+ *
+ */
+
+#include <arm_neon.h>
+#include <cassert>
+#include <cstdint>
+#include <mask_neon.h>
+
+#define TRANSPOSE_FP16_4x4(row0, row1, row2, row3)                             \
```

Comment:
It is recommended to wrap the multi-sentense expression in #define to prevent unwanted behaviors:

```
#define TRANSPOSE_FP16_4x4(row0, row1, row2, row3) \
do { \
  your sentences ... \
} whiile(0)
```

Reply:
Applied the change! Thanks!